### ISSUE TITLE
[Bug] 缺少全域隨機 seed 控制入口，整個框架訓練無法重現

### ISSUE THREAD

**Issue by s9213712** [s9213712/minicnn#41]
## 問題描述

minicnn 框架中沒有任何全域 random seed 設定的機制。多個地方使用了隨機性：

1. `src/minicnn/ops/nn_ops.py:dropout()` — `np.random.random()`（全域狀態）
2. `src/minicnn/nn/layers.py:Linear.__init__()` — `np.random.default_rng()`（無 seed）
3. `src/minicnn/nn/layers.py:Conv2d.__init__()` — `np.random.default_rng()`（無 seed）
4. `src/minicnn/data/cifar10.py` — 資料集 shuffle
5. `src/minicnn/training/train_autograd.py` — 各種隨機操作

所有這些在每次執行時都使用不同的隨機狀態，導致：
- 相同 config 的兩次訓練無法得到相同結果
- Bug 報告難以重現（「在我的機器上 val_acc 是 88%，你的是 85%？」）
- 論文級實驗需要多次運行求均值，但無法確保各次運行唯一差異是模型本身

`config/schema.py` 中有 `dataset_seed`、`init_seed`、`train_seed` 三個 seed 欄位，但它們實際上是否被使用未知（schema 定義了，但執行時是否傳遞需確認）。

## 建議修復

在 CLI / `train_from_config()` 入口點設定全域 seed：

```python
def set_global_seed(seed: int):
    import numpy as np
    import random as py_random
    py_random.seed(seed)
    np.random.seed(seed)
    # 若使用 torch：
    # import torch; torch.manual_seed(seed)
```

並在 `ExperimentConfig.train.init_seed` 被讀取後立即呼叫。

## 相關位置

- `src/minicnn/config/schema.py:42-44` — `dataset_seed`, `init_seed`, `train_seed` 定義
- `src/minicnn/ops/nn_ops.py:53` — dropout 使用全域 RNG
- `src/minicnn/nn/layers.py` — 層初始化使用 `default_rng()`（無 seed）
